### ISSUE TITLE
tzname() inconsistent between tzwin() and tzwinlocal() and neither reflect actual DST.

### ISSUE THREAD

**Issue by labrys** [dateutil/dateutil#143]
Time zone names are inconsistent between `tzwin()` and `tzwinlocal()` and neither reflect actual DST.

Tested on Windows for all US time zones with current master SHA1: b46e721 and for proposed fix at #142 - both fail.

```python
import datetime
import dateutil.tz

tw = dateutil.tz.tzwin('Eastern Standard Time')
twl = dateutil.tz.tzwinlocal()
tests = [('EDT should be started', '2015-03-08 03:01'),
         ('Still not EDT', '2015-03-15 01:59'),
         ('tzwin is EDT', '2015-03-15 02:00'),
         ('tzwinlocal still not EDT', '2015-03-16 23:59'),
         ('Finally EDT!', '2015-03-17 00:00'),
         ('EDT should have ended', '2015-11-01 03:01'),
         ('Still not EST', '2015-11-08 01:59'),
         ('tzwin is EST', '2015-11-08 02:00'),
         ('tzwinlocal still not EST', '2015-11-15 23:59'),
         ('Finally EST!', '2015-11-16 00:00')]

row = '{:<16}  |  {:<21}  |  {:<21}  |  {}'
print row.format('Date', 'tzwin', 'tzwinlocal', 'Notes')
print '{:-<100}'.format('')
for each in tests:
    desc, date = each
    test = datetime.datetime.strptime(date, '%Y-%m-%d %H:%M')
    tw_test = test.replace(tzinfo=tw)
    twl_test = test.replace(tzinfo=twl)
    print row.format(date, tw_test.tzname(), twl_test.tzname(), desc)
```
result:
```
Date              |  tzwin                  |  tzwinlocal             |  Notes
----------------------------------------------------------------------------------------------------
2015-03-08 03:01  |  Eastern Standard Time  |  Eastern Standard Time  |  EDT should be started
2015-03-15 01:59  |  Eastern Standard Time  |  Eastern Standard Time  |  Still not EDT
2015-03-15 02:00  |  Eastern Daylight Time  |  Eastern Standard Time  |  tzwin is EDT
2015-03-16 23:59  |  Eastern Daylight Time  |  Eastern Standard Time  |  tzwinlocal still not EDT
2015-03-17 00:00  |  Eastern Daylight Time  |  Eastern Daylight Time  |  Finally EDT!
2015-11-01 03:01  |  Eastern Daylight Time  |  Eastern Daylight Time  |  EDT should have ended
2015-11-08 01:59  |  Eastern Daylight Time  |  Eastern Daylight Time  |  Still not EST
2015-11-08 02:00  |  Eastern Standard Time  |  Eastern Daylight Time  |  tzwin is EST
2015-11-15 23:59  |  Eastern Standard Time  |  Eastern Daylight Time  |  tzwinlocal still not EST
2015-11-16 00:00  |  Eastern Standard Time  |  Eastern Standard Time  |  Finally EST!
```

**Comment by pganssle** [comment]
How very strange. Really good find.

By the way, if you don't mind, in the future can you use the print function syntax? `print(x)` instead of `print x`. I'm often running these tests on Python 3.x so it makes it more difficult to just copy+paste.

**Comment by pganssle** [comment]
OK, there are two problems:

```python
import dateutil.tz

def print_transitions(tzwi):
    trstd = (tzwi._stdmonth,
             tzwi._stddayofweek,   # Sunday = 0
             tzwi._stdweeknumber,  # Last = 5
             tzwi._stdhour,
             tzwi._stdminute)
    trdst = (tzwi._dstmonth,
             tzwi._dstdayofweek,   # Sunday = 0
             tzwi._dstweeknumber,  # Last = 5
             tzwi._dsthour,
             tzwi._dstminute)

    print("Standard Transition:")
    print(trstd)
    print("DST Transition:")
    print(trdst)


tw = dateutil.tz.tzwin('Eastern Standard Time')
twl = dateutil.tz.tzwinlocal()

print("TZWin:")
print_transitions(tw)       # This is correct
print("TZWinLocal:")
print_transitions(twl)      # Says it's actually the 2nd TUESDAY of March

print(dateutil.tz.win.picknthweekday(2015, tw._dstmonth, tw._dstdayofweek,
                                     tw._dsthour, tw._dstminute,
                                     tw._dstweeknumber))
```

This returns:
```
TZWin:
Standard Transition:
(11, 0, 1, 2, 0)
DST Transition:
(3, 0, 2, 2, 0)
TZWinLocal:
Standard Transition:
(11, 1, 2, 0, 0)
DST Transition:
(3, 2, 2, 0, 0)
2015-03-15 02:00:00
```

The first problem is that `tzwinlocal` is pulling the keys "DaylightStart" and "StandardStart", which evidently have a different format than `tzwin` is expecting.

Assuming that it's expecting a 1-based index for the "nth day of week", I think this will fix it:

```python
def picknthweekday(year, month, dayofweek, hour, minute, whichweek):
    """dayofweek == 0 means Sunday, whichweek 5 means last instance"""
    first = datetime.datetime(year, month, 1, hour, minute)
    weekdayone = first.replace(day=((dayofweek-first.isoweekday()) % 7+1))
    wd = weekdayone + ((whichweek-1) * ONEWEEK)
    if (wd.month != month):
        wd -= ONEWEEK

    return wd
```

I'll have to look into the formats of the Windows registry keys to be sure though.

**Comment by pganssle** [comment]
OK, it turns out that the `tzwinlocal()` issue is due to the fact that for whatever reason, in "SYSTEM\CurrentControlSet\Control\TimeZoneInformation" (the local timezone key), DaylightStart and StandardStart don't use the normal ordering for the `SYSTEMTIME` struct. This should be fixed in #178.

**Comment by labrys** [comment]
Confirmed working on Windows with various versions of python.
